storage/zynq7000-flash/zynq7000-flash.c

```diff
@@ -0,0 +1,529 @@
+		part = strg->parent->parts;
+		if (part != NULL) {
+			do {
+				partID++;
+				part = part->next;
+			} while (part != strg->parent->parts);
+		}
+	}
+
+	/* Add mtdchar device */
+	if (strg->dev->mtd != NULL) {
+		oid->id &= ~MTD_MASK;
+		oid->id |= MTD_CHAR;
+
+		res = snprintf(path, sizeof(path), "/dev/mtd%u", strg->dev->id);
+		if (strg->parent != NULL) {
+			snprintf(path + res, sizeof(path), "p%u", partID);
+		}
+
+		res = create_dev(oid, path);
+		if (res < 0) {
+			fprintf(stderr, "zynq7000-flash: failed to create a device file, err: %d\n", res);
+			return res;
+		}
+	}
+
+	/* Add mtdblock device */
+	if (strg->dev->blk != NULL) {
+		oid->id &= ~MTD_MASK;
+		oid->id |= MTD_BLOCK;
+
+		res = snprintf(path, sizeof(path), "/dev/mtdblock%u", strg->dev->id);
+		if (strg->parent != NULL) {
+			snprintf(path + res, sizeof(path), "p%u", partID);
+		}
+
+		res = create_dev(oid, path);
+		if (res < 0) {
+			fprintf(stderr, "zynq7000-flash: failed to create a device file, err: %d\n", res);
+			return res;
+		}
+	}
+
+	return EOK;
+}
+
+
+static int flash_partAdd(const char *parentPath, off_t start, size_t size)
+{
+	int err;
+	oid_t poid, oid;
+	storage_t *strg, *parent;
+
+	err = flash_oidResolve(parentPath, &poid);
+	if (err < 0) {
+		printf("zynq7000-flash: cannot resolve %s\n", parentPath);
+		return err;
+	}
+
+	parent = storage_get(GET_STORAGE_ID(poid.id));
+	if (parent == NULL) {
+		err = -EINVAL;
+		fprintf(stderr, "zynq7000-flash: failed to find a parent %s, err: %d\n", parentPath, err);
+		return err;
+	}
+
+	strg = malloc(sizeof(storage_t));
+	if (strg == NULL) {
+		err = -ENOMEM;
+		fprintf(stderr, "zynq7000-flash: failed to allocate a device, err: %d\n", err);
+		return err;
+	}
+
+	strg->start = start;
+	strg->size = size;
+	strg->parent = parent;
+	strg->dev = parent->dev;
+	strg->parts = NULL;
+
+	err = storage_add(strg, &oid);
+	if (err < 0) {
+		free(strg);
+		fprintf(stderr, "zynq7000-flash: failed to create a partition, err: %d\n", err);
+		return err;
+	}
+
+	err = flash_createMtdDev(strg, &oid);
+	if (err < 0) {
+		storage_remove(strg);
+		free(strg);
+		return err;
+	}
+
+	return GET_STORAGE_ID(oid.id);
+}
+
+
+static int flash_optsParse(int argc, char **argv)
+{
+	int err, c;
+	unsigned int id;
+	oid_t oid;
+	offs_t start;
+	size_t size;
+	char *devPath, *arg, *fs;
+
+	while ((c = getopt(argc, argv, "p:r:h")) != -1) {
+		err = -EINVAL;
+		switch (c) {
+			case 'p': /* <dev:start:size> */
+				devPath = optarg;
+				arg = strchr(optarg, ':');
+				if (!arg) {
+					fprintf(stderr, "zynq7000-flash: missing a partition offset, err: %d\n", err);
+					return err;
+				}
+
+				*arg++ = '\0';
+				start = strtol(arg, &arg, 0);
+				if (*arg++ != ':') {
+					fprintf(stderr, "zynq7000-flash: missing a partition size, err: %d\n", err);
+					return err;
+				}
+
+				size = strtol(arg, &arg, 0);
+				if (*arg != '\0') {
+					fprintf(stderr, "zynq7000-flash: wrong arguments %s, err: %d\n", arg, err);
```

Comment:
Such a message will be more readable.
```suggestion
					fprintf(stderr, "zynq7000-flash: wrong partition size\n");
```
Is there a need to print the error number (-EINVAL) in the messages above?

Reply:
Ok, I have changed the message. In general, each of the log message contains the error number. I would not make an exception in this case.